class-gf-razorpay.php

```diff
@@ -0,0 +1,285 @@
+            array('name' => 'address2', 'label' => 'Address 2', 'meta_name' => 'billingInformation_address2'),
+            array('name' => 'city', 'label' => 'City', 'meta_name' => 'billingInformation_city'),
+            array('name' => 'state', 'label' => 'State', 'meta_name' => 'billingInformation_state'),
+            array('name' => 'zip', 'label' => 'Zip', 'meta_name' => 'billingInformation_zip'),
+            array('name' => 'country', 'label' => 'Country', 'meta_name' => 'billingInformation_country'),
+        );
+    }
+
+    public function callback()
+    {
+        $entry_id = $_COOKIE['entry_id'];
+
+        $razorpay_order_id = $_COOKIE['razorpay_order_id'];
+
+        $entry = GFAPI::get_entry($entry_id);
+
+        $api = new Api($this->get_plugin_setting(self::GF_RAZORPAY_KEY),
+            $this->get_plugin_setting(self::GF_RAZORPAY_SECRET));
+
+        $attributes = array (
+            'razorpay_order_id'   => $razorpay_order_id,
+            'razorpay_payment_id' => rgpost('razorpay_payment_id'),
+            'razorpay_signature'  => rgpost('razorpay_signature'),
+        );
+
+        $success = false;
+
+        if ($entry and !empty(rgpost('razorpay_payment_id')) and !empty(rgpost('razorpay_signature')))
+        {
+            try
+            {
+                $api->utility->verifyPaymentSignature($attributes);
+                $success = true;
+            }
+            catch (Errors\SignatureVerificationError $e)
+            {
+                $error = $e->getMessage();
+            }
+        }
+        else
+        {
+            $error = 'Payment Failed';
+        }
+
+        if ($success === true)
+        {
+            $action['id']             = $attributes['razorpay_payment_id'];
+            $action['type']           = 'complete_payment';
+            $action['transaction_id'] = $attributes['razorpay_payment_id'];
+            $action['amount']         = $entry['payment_amount'];
+            $action['entry_id']       = $entry['id'];
+        }
+        else
+        {
+            $action['id']             = $attributes['razorpay_payment_id'];
+            $action['type']           = 'fail_payment';
+            $action['transaction_id'] = $attributes['razorpay_payment_id'];
+            $action['entry_id']       = $entry['id'];
+            $action['amount']         = $entry['payment_amount'];
+        }
+
+        return $action;
+    }
+
+    public function post_callback($callback_action, $callback_result) {
+        $entry               = GFAPI::get_entry($callback_action['entry_id']);
+        $feed                = $this->get_payment_feed( $entry );
+        $transaction_id      = rgar($callback_action, 'transaction_id');
+        $amount              = rgar($callback_action, 'amount');
+
+        do_action('gform_razorpay_post_payment_' . $callback_action['type'],  $_POST, $entry, $feed);
+    }
+
+    public function generate_razorpay_form($entry, $customer_fields, $form)
```

Comment:
i dont think we should do that. This function is handling the full payment. Breaking down in methods would confuse people. generate_order_form is already in other method